source/docs/advanced/features/external-logins.md

```diff
@@ -0,0 +1,314 @@
+import customConfigProvider from "path_to_your_customConfigProvider";
+import { registerExternalLoginProvider } from "auth-external-login/server/providers";
+
+configService.append(customConfigProvider);
+
+const makeCustomProvider = (request, externalLoginHandler, callbackUrl) => {
+  const providerConfig = request.config.customProviderConfig;
+  if(!providerConfig.clientId || !providerConfig.clientSecret) {
+    throw new Error(`Custom provider is not properly configured. clientId and clientSecret are required!`);
+  }
+  const shopUrl = getCurrentShopConfig(req.config).url; // frequently needed to keep current store selected upon redirection
+  return new CustomProvider(externalLoginHandler, providerConfig, {
+    shopUrl,
+    callbackUrl,
+  });
+};
+
+registerExternalLoginProvider("custom", makeCustomProvider);
+```
+
+#### The LoginProviderInerface
+
+To make creating a provider easier, `LoginProviderInerface` can be implemented. By implementing `LoginProviderInerface` you have to implement the 2 aforementioned middleware and you can pick what inputs your constructor takes. You will then register the maker function mentioned above to extract the data needed by your provider and return an instance of it.
+
+#### The LoginProviderBase
+
+By implementing the Google and Facebook providers using passport these patterns emerged:
+* The `loginCallbackMiddleware` is just a call to the external login handler's `login` function.
+* The provider commonly needs:
+  * The login handler
+  * The provider's own config
+  * The shop and the callback URLs
+
+`LoginProviderBase` is a simple implementation of the `LoginProviderInerface` that has the constructor (see below for signature) and the `loginCallbackMiddleware` implemented.
+
+```
+  constructor(externalLoginHandler, providerConfig, { shopUrl, callbackUrl }) {
+    ...
+  }
+```
+
+What is left is implementing `requestLoginMiddleware` which is easily doable using passport:
+
+```
+requestLoginMiddleware(req, res, next) {
+
+  const passportAuthenticator = new Passport();
+
+  passportAuthenticator.use(
+    new FacebookStrategy(
+      {
+        clientID: this.providerConfig.clientId,
+        clientSecret: this.providerConfig.clientSecret,
+        callbackURL: `${this.shopUrl}${this.callbackUrl}`,
+        profileFields: ["email", "name"],
+      },
+      (_, __, profile, cb) => {
+        const {
+          email,
+          first_name: firstname,
+          last_name: lastname,
+        } = profile._json;
+
+        if (!(email && firstname && lastname)) {
+          cb(new MissingInformationError());
+        } else {
+          cb(null, new ExternalLoginUser(email, firstname, lastname));
+        }
+      }
+    )
+  );
+
+  passportAuthenticator.authenticate("facebook", {
+    failureRedirect: `${this.shopUrl}/login`,
+    session: false,
+    scope: ["email"],
+  })(req, res, next);
+}
+```
+
+### Creating your own ExternalLoginHandler
```

Comment:
not sure anybody but us would want to do this

Reply:
Proximis/BigCommerce doesnt have these implemented so they would need to build them themselves

Reply:
@mo-fc this is in my opinion the bad reason, it should be our responsibility. 

But this may be handy in some cases using an external user directory.